module/zfs/zil.c

```diff
@@ -1253,6 +1254,24 @@ zil_lwb_write_done(zio_t *zio)
 	 * written out.
 	 */
 	if (zio->io_error != 0) {
+		/*
+		 * Copy the write error to zcw, becaues the zil_lwb_write_done
+		 * error is not propagated to zil_lwb_flush_vdevs_done, which
+		 * will cause zil_commit_impl to return without committing
+		 * the data.
+		 * Refer https://github.com/openzfs/zfs/issues/12391
```

Comment:
1. `typo` in `becaues`
2. I find the comment here more confusing than useful. I'd prefer a comment in `zil_commit_impl` in the place where we check `zcw_zio_error` that explains the entire error propagation path (for both the `write` and `flush` `done` callbacks)

Reply:
@problame thank you for the comments. I have updated the PR.